add a g2p example

iclementine opened the pull request:
<!-- Demo: https://github.com/PaddlePaddle/PaddleNLP/pull/26 -->
### PR types
<!-- One of [ New features | Bug fixes | Function optimization | Performance optimization | Breaking changes | Others ] -->
Others

### PR changes
<!-- One of [ Models | APIs | Docs | Others ] -->
Others

### Describe
<!-- Describe what this PR does -->
Add an example to convert transcriptions from Baker dataset into pinyin.

iclementine committed:
add an example to convert transcription to pinyin with pypinyin and jieba

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/PaddlePaddle/DeepSpeech?pullRequest=623) <br/>All committers have signed the CLA.

iclementine committed:
format code

zh794390558 commented on examples/chinese_g2p/requirements.txt:
```diff
@@ -0,0 +1,2 @@
+jieba
+pypinyin
```
pypinyin可以不用安装了。主目录的 setup.sh 会安装。

iclementine replied:
已经修改

zh794390558 commented on examples/chinese_g2p/data/README.md:
```diff
@@ -0,0 +1,3 @@
+# Download Baker dataset
```
README.md放到上一级目录，data目录可以删除。download脚本或run.sh 创建 data 目录。

iclementine replied:
okay，虽然这个数据集还是需要手动下载

zh794390558 commented on examples/chinese_g2p/local/data_download.sh:
```diff
@@ -0,0 +1,21 @@
+
+. ${MAIN_ROOT}/utils/utility.sh
+
+DOWNLOAD_DIR=$(dirname $0)/../data
```
DOWNLOAD_DIR 可以参数传入。

zh794390558 replied:
source ${MAIN_ROOT}/utils/parse_options.sh 这个可以用来解析参数。

iclementine replied:
okay

zh794390558 commented on examples/chinese_g2p/local/data_download.sh:
```diff
@@ -0,0 +1,21 @@
+. ${MAIN_ROOT}/utils/utility.sh
+
+DOWNLOAD_DIR=$(dirname $0)/../data
+mkdir -p ${DOWNLOAD_DIR}
```
这部分放到run.sh 可以会好些。

iclementine replied:
已经修改

zh794390558 commented on examples/chinese_g2p/local/extract_pinyin.py:
```diff
@@ -0,0 +1,33 @@
+from pypinyin import lazy_pinyin, Style
+
+def extract_pinyin(source, target, use_jieba=False):
+    with open(source, 'rt', encoding='utf-8') as f:
```
f -> fin, g->fout

iclementine replied:
已经修改

zh794390558 commented on examples/chinese_g2p/run.sh:
```diff
@@ -0,0 +1,37 @@
+ARCHIVE="data/BZNSYP.rar"
+
+echo "Extracting Prosody Labeling"
+LABEL_FILE='ProsodyLabeling/000001-010000.txt'
```
变量放到一起，都小写即可。

iclementine replied:
已经修改

iclementine committed:
1. remove script for data downloading, since Baker dataset is not easily downloaded via terminal;
2. remove pypinyin as an extra requirement; it is alreay required by the main project;
3. clean code.

zh794390558 commented on examples/chinese_g2p/local/extract_pinyin.py:
```diff
@@ -0,0 +1,54 @@
+                        style=Style.TONE3,
+                        neutral_tone_with_five=True)
+                    transcription = ' '.join(syllables)
+                    fout.write(f'\t{transcription}\n')
```
输出文件也保持： utt transcript格式。

iclementine replied:
之所以这么做是为了和数据集中的 Prosody 一致方便 diff.
是要改成什么样子呢？

zh794390558 replied:
按照 utt transcript格式即可，对比diff用vimdiff工具来比。

zh794390558 commented on examples/chinese_g2p/README.md:
```diff
@@ -0,0 +1,3 @@
+# Download Baker dataset
+
+Baker dataset has to be downloaded mannually and moved to 'data/', because you will have to pass the CATTCHA from a browswe to download the dataset.
```
给个数据集的地址。

iclementine replied:
done

zh794390558 commented on examples/chinese_g2p/local/prepare_dataset.sh:
```diff
@@ -0,0 +1,31 @@
+label_file='ProsodyLabeling/000001-010000.txt'
+filename='000001-010000.txt'
+unrar e ${archive} ${label_file}
+mv ${filename} ${exp_dir}
```
copy not mv

iclementine replied:
unrar 这样子默认是解压到当前目录。但是目的是解压到 exp_dir, 为了不留下残余的文件，所以 mv。话说有办法直接解压到 exp_dir 吗？

zh794390558 replied:
解压到data是正常的。copy一份过去就行。

iclementine replied:
如果在这个 example 的目录执行 run.sh 会解压到当前目录，而不是 data 目录。

zh794390558 replied:
unrar e test.rar 解压文件到当前目录

  unrar x test.rar /path/to/extract

Commit:
change output format

zh794390558 reviewed (approved):
LGTM
